ballista/scheduler/src/scheduler_server/query_stage_scheduler.rs

```diff
@@ -119,44 +119,59 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan>
 
                 let state = self.state.clone();
                 tokio::spawn(async move {
-                    let event = if let Err(e) = state
-                        .submit_job(&job_id, &job_name, session_ctx, &plan, queued_at)
-                        .await
-                    {
-                        let fail_message = format!("Error planning job {job_id}: {e:?}");
-                        error!("{}", &fail_message);
-                        QueryStageSchedulerEvent::JobPlanningFailed {
-                            job_id,
-                            fail_message,
-                            queued_at,
-                            failed_at: timestamp_millis(),
-                        }
-                    } else {
-                        QueryStageSchedulerEvent::JobSubmitted {
-                            job_id,
-                            queued_at,
-                            submitted_at: timestamp_millis(),
-                            resubmit: false,
-                        }
-                    };
+                    let event =
+                        match state.plan_job(&job_id, session_ctx.clone(), &plan).await {
+                            Ok(plan) => QueryStageSchedulerEvent::JobSubmitted {
+                                job_id,
+                                job_name,
+                                session_id: session_ctx.session_id(),
+                                queued_at,
+                                submitted_at: timestamp_millis(),
+                                resubmit: false,
+                                plan,
```

Comment:
move the job related status updating to `QueryStageSchedulerEvent::JobSubmitted`